File: demo/vulnerable_demo.js
// Demo: Vulnerable Code for Agent Security Scanner

// 1. XSS - innerHTML vulnerability
function displayMessage(userInput) {
  document.getElementById('output').innerHTML = userInput;
}

// 2. SQL Injection - template literal
function getUser(userId) {
  return db.query(`SELECT * FROM users WHERE id = ${userId}`);
}

// 3. SQL Injection - simple concatenation
function deleteUser(id) {
  return db.query("DELETE FROM users WHERE id = " + id);
}

module.exports = { displayMessage, getUser, deleteUser };

File: demo/vulnerable_demo.py
# Demo: Vulnerable Python Code for Agent Security Scanner

import sqlite3

# 1. SQL Injection - f-string (should be fixed now!)
def get_user_by_id(user_id):
    cursor.execute(f"SELECT * FROM users WHERE id = {user_id}")
    return cursor.fetchone()

# 2. SQL Injection - .format()
def search_products(term):
    cursor.execute("SELECT * FROM products WHERE name = '{}'".format(term))
    return cursor.fetchall()

# 3. SQL Injection - % formatting
def get_order(order_id):
    cursor.execute("SELECT * FROM orders WHERE id = %s" % (order_id,))
    return cursor.fetchone()

# 4. SQL Injection - simple concatenation
def delete_user(user_id):
    cursor.execute("DELETE FROM users WHERE id = " + user_id)
